### PR TITLE
Ensure zstd decompressor is always closed, even if clients do not send a FinishWrite request

### PR DESCRIPTION
**Pull request by bduffany**
---

**Version bump**: Patch

### CODE REVIEW

File: server/remote_cache/byte_stream_server/byte_stream_server.go [hunk 1]
```diff
@@ -259,6 +260,11 @@ func (w *writeState) Write(buf []byte) error {
 }
 
 func (w *writeState) Close() error {
+	if w.closed {
+		return nil
+	}
+	w.closed = true
+
```
Comment (tylerwilliams) [inline server/remote_cache/byte_stream_server/byte_stream_server.go:267]: could you nil out w.decompressorCloser after closing it? and remove this block?

Comment (bduffany) [reply server/remote_cache/byte_stream_server/byte_stream_server.go]: lol, I had that before, but was on the fence. Changed it back

Comment (tylerwilliams) [reply server/remote_cache/byte_stream_server/byte_stream_server.go]: it's somewhat idiomatic in go to do that afaik. Don't have a source handy tho, lmk if you find one.

**Review by tylerwilliams — APPROVED** [review]
LGTM

**Pull request merged**